### PR TITLE
[netcore] Guard for skipFrames overflow, update test exclusions

### PR DESCRIPTION
**Pull request by filipnavara**

### CODE REVIEW

File: mcs/class/System.Private.CoreLib/System.Diagnostics/StackTrace.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ void InitializeForCurrentThread (int skipFrames, bool needFileInfo)
 
 			StackFrame sf;
 			var frames = new List<StackFrame> ();
-			while ((sf = new StackFrame (skipFrames, needFileInfo)) != null && sf.GetMethod () != null) {
+			while (skipFrames >= 0 && (sf = new StackFrame (skipFrames, needFileInfo)) != null && sf.GetMethod () != null) {
```
Comment (filipnavara) [inline mcs/class/System.Private.CoreLib/System.Diagnostics/StackTrace.cs]: Note that the `new StackFrame(...) != null` doesn't make sense, but it was already in the code before.

Comment (marek-safar) [reply mcs/class/System.Private.CoreLib/System.Diagnostics/StackTrace.cs]: Could you please fix that as well

Comment (filipnavara) [reply mcs/class/System.Private.CoreLib/System.Diagnostics/StackTrace.cs]: Done

**Comment by marek-safar** [comment]
Thank you!

**Pull request merged**